File: src/main/java/com/example/venka/demo/service/MessageService.java
package com.example.venka.demo.service;

import com.example.venka.demo.model.Message;
import org.bson.types.ObjectId;

import java.util.List;

public interface MessageService {

    void insert(final Message message);

    void delete(final ObjectId id);

    List<Message> getAll();

    Message getById(final ObjectId id);
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/service/MessageService.kt
package com.example.venka.kotlinspringsparkdemo.service

import com.example.venka.kotlinspringsparkdemo.model.Message
import org.bson.types.ObjectId

interface MessageService {

    val all: List<Message>

    fun insert(message: Message)

    fun delete(id: ObjectId)

    fun getById(id: ObjectId): Message
}
